src/libstd/io/util.rs

```diff
@@ -44,8 +44,7 @@ pub fn copy<R: ?Sized, W: ?Sized>(reader: &mut R, writer: &mut W) -> io::Result<
     where R: Read, W: Write
 {
     let mut buf = unsafe {
-        #[allow(deprecated)]
-        let mut buf: [u8; super::DEFAULT_BUF_SIZE] = mem::uninitialized();
+        let mut buf: [u8; super::DEFAULT_BUF_SIZE] = mem::MaybeUninit::uninit().assume_init();
```

Comment:
I'm afraid this is not the right fix. You are creating uninitialized integers, which under our current rules is UB.

Reply:
Thanks for the feedback. I'll try to understand more deeply and fix.